src/platform/user/profile/utilities/index.js

```diff
@@ -141,3 +141,78 @@ export function teardownProfileSession() {
   sessionStorage.removeItem('shouldRedirectExpiredSession');
   clearSentryLoginType();
 }
+
+export const getValidationMessageKey = (
+  suggestedAddresses,
+  validationKey,
+  addressValidationError,
+) => {
+  let validationErrorKey;
+  const singleSuggestion = suggestedAddresses.length === 1;
+  const containsBadUnitNumber =
+    suggestedAddresses.filter(
+      address =>
+        address.addressMetaData?.deliveryPointValidation ===
+        'STREET_NUMBER_VALIDATED_BUT_BAD_UNIT_NUMBER',
+    ).length > 0;
+
+  const containsMissingUnitNumber =
+    suggestedAddresses.filter(
+      address =>
+        address.addressMetaData?.deliveryPointValidation ===
+        'STREET_NUMBER_VALIDATED_BUT_MISSING_UNIT_NUMBER',
+    ).length > 0;
+
+  if (singleSuggestion && containsBadUnitNumber) {
+    validationErrorKey = validationKey
+      ? 'badUnitNumberOverride'
+      : 'badUnitNumber';
+  }
+
+  if (singleSuggestion && containsMissingUnitNumber) {
+    validationErrorKey = validationKey
+      ? 'missingUnitNumberOverride'
+      : 'missingUnitNumber';
+  }
+
+  if (
+    singleSuggestion &&
+    !addressValidationError &&
+    !containsMissingUnitNumber &&
+    !containsBadUnitNumber
+  ) {
+    validationErrorKey = validationKey
+      ? 'showSuggestionsOverride'
+      : 'showSuggestions';
+  }
+
+  if (!singleSuggestion && !addressValidationError) {
+    validationErrorKey = validationKey
+      ? 'showSuggestionsOverride'
+      : 'showSuggestions';
+  }
+
+  if (addressValidationError) {
+    validationErrorKey = 'validationError';
+  }
+
+  return validationErrorKey;
+};
+
+export const showAddressValidationModal = suggestedAddresses => {
+  if (
+    suggestedAddresses.length === 1 &&
+    (suggestedAddresses[0].addressMetaData?.deliveryPointValidation ===
+      'STREET_NUMBER_VALIDATED_BUT_BAD_UNIT_NUMBER' ||
+      suggestedAddresses[0].addressMetaData?.deliveryPointValidation ===
+        'STREET_NUMBER_VALIDATED_BUT_MISSING_UNIT_NUMBER')
+  ) {
+    return true;
+  }
+
+  if (suggestedAddresses.length > 1) {
+    return true;
+  }
+
+  return suggestedAddresses[0]?.addressMetaData?.confidenceScore < 80;
```

Comment:
I feel like this should be part of the first check at the top of this function (if there is only a single suggestion)

Reply:
its done that way instead of this:
```
if (suggestedAddresses[0]?.addressMetaData?.confidenceScore < 80) {
  return true;
}
return false;
```
and I wanted it broken out from the bad / missing unit number stuff incase that changes independently